backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -1237,10 +1361,29 @@ public ExperimentRunPaginationDTO findExperimentRuns(FindExperimentRuns queryPar
       experimentRunRoot.alias("exp");
       List<Predicate> finalPredicatesList = new ArrayList<>();
 
+      List<String> projectIds = new ArrayList<>();
       if (!queryParameters.getProjectId().isEmpty()) {
-        Expression<String> exp = experimentRunRoot.get(ModelDBConstants.PROJECT_ID);
-        Predicate predicate2 = builder.equal(exp, queryParameters.getProjectId());
-        finalPredicatesList.add(predicate2);
+        projectIds.add(queryParameters.getProjectId());
+      } else {
+        List<String> workspaceProjectIDs =
+            projectDAO.getWorkspaceProjectIDs(
+                queryParameters.getWorkspaceName(), currentLoginUserInfo);
+        if (workspaceProjectIDs == null || workspaceProjectIDs.isEmpty()) {
+          LOGGER.warn(
+              "accessible project for the experimentRuns not found for given workspace : {}",
+              queryParameters.getWorkspaceName());
+          ExperimentRunPaginationDTO experimentRunPaginationDTO = new ExperimentRunPaginationDTO();
+          experimentRunPaginationDTO.setExperimentRuns(Collections.emptyList());
+          experimentRunPaginationDTO.setTotalRecords(0L);
+          return experimentRunPaginationDTO;
+        }
+        projectIds.addAll(workspaceProjectIDs);
+      }
+
+      if (!projectIds.isEmpty()) {
```

Comment:
if projectIds isEmpty what does it mean? can we return early? here it seems if there is no project accessible in workspace we search across all the projects, which is wrong.

Reply:
Also if you think this is causing https://vertaai.atlassian.net/browse/VR-4091 please assign it to yourself.

Reply:
@ravishetye this if is only for the safety cause because if the workspace has no project then it will return from https://github.com/VertaAI/modeldb/pull/568/files#diff-1a3287003b54e5e5b064b79266c3d82bR1371